test/deps.sh

```diff
@@ -17,6 +19,11 @@ name () {
 test () {
   echo "## TESTING DEPENDENCY: $1"
   pushd node_modules/$1
+  npm install --only=dev
+  pushd node_modules
+  # remove duplicates of upstream deps
```

Comment:
@dominictarr 

Have you already looked at the code? I think this resolves the problem you've illustrated, but if it doesn't I'd love to understand why it won't work.

Reply:
oh I was reacting to "dynamic" and didn't see the "# remove duplicates of upstream deps" comment.
I must say I don't really understand how it works... it looks more like it deletes the production after each test run?

Oh so it gets the production deps used by `ssb-server`, then it cd's into a dev dep, then it installs the dev deps, then it deletes anything which _also_ happens to be a product dep of `ssb-server`, so only extra needed dev deps remain. Hmm, so I think this does address my concern that duplicate deps will be created...

but I honestly think this is not as nice as just putting those extra deps into `ssb-server`'s devdeps.
You campaigned for a long time to get ssb-server to use a lockfile, now it does - but this abandons the lockfile for the dev deps! If we are gonna do a lock file _everything should be in the lockfile_

Reply:
how do you feel about a script that automatically includes them in the dev deps?
I am a lockfile _all-or-nothing-ist_

sure, the dev deps appearing unused is an easy mistake, but that's just needs a little documentation about how the integration tests work.

Reply:
btw, the dev deps are the things needed to do development - i.e. run the tests. because of running integration tests, they _are_ also our dev deps.

Reply:
We already have a script that specifies our dependencies in a lockfile: `npm install`.

If you're interested in specifying plugin devdeps in a lockfile then I'd propose that we add a lockfile to the plugin repos. This would resolve your concern *and* make those repos more conventional, which increases accessibility.

Would you be open to pull requests adding lockfile to each of the plugin repos?

Reply:
@christianbundy I suggested a compromise that I thought would work for both of use and asked how you felt about it - please work with my by at least describing what your problems are with it, so as to show me openings towards something else that might work.

I do not agree that lockfiles to each of those projects is the right solution. The point is to test those modules with `ssb-server`'s deps, not with the deps specified when `ssb-gossip` or what have you's deps. (the plugins deps are probably old, I don't want to go and publish new versions of all the plugins just because one has been updated)

This is a weird move, but it's also about accessability. I want the test suite to have very good coverage so that you can be confidant that the PR etc you've made doesn't break anything. (people are afraid to break stuff, including me)